### PR TITLE
Bugfix download images

### PR DESCRIPTION
**Pull request by a-cool-train**
*Description of changes:*
- Currently Curated Packages images are not getting fetched. This could potentially cause an issue when creating a cluster during disconnected scenarios
- This PR includes the extraction of images from curated packages bundles.
- Temporary Solution: Currently private curated packages don't have a root level project hence this PR adds a root path to the images of the packages
- Currently the images of curated packages are referenced through digest, this PR removes the @ included in the digest
- Temporary Solution: Currently curated packages are in a private account, this PR adds constants to use for getting the images from the private account

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jaxesn — COMMENTED** [review]
These are for the curated packages images themselves? Will this only trigger if the user is running with a valid license?  I just want to make sure we arent breaking import/download images for uses without enterprise and therefore no access to the private ecr repos

File: pkg/docker/registry.go [hunk 1]
```diff
@@ -3,10 +3,21 @@ package docker
...
 )
 
+// These constants are temporary since currently there is a limitation on harbor
+// Harbor requires root level projects but curated packages private account currently
+// doesn't have support for root level
+const (
+	packageProdLocation = "783794618700.dkr.ecr.us-west-2.amazonaws.com"
+	packageDevLocation  = "857151390494.dkr.ecr.us-west-2.amazonaws.com"
+	publicProdECRName   = "eks-anywhere"
+	publicDevECRName    = "l0g8r8j6"
```
Comment (g-gaston) [inline pkg/docker/registry.go:18]: could we just use the same "folder" for both prod and dev? I would vote for `eks-anywhere`

File: pkg/executables/docker.go [hunk 2]
```diff
@@ -80,7 +84,9 @@ func (d *Docker) CgroupVersion(ctx context.Context) (int, error) {
 }
 
 func (d *Docker) TagImage(ctx context.Context, image string, endpoint string) error {
-	localImage := strings.ReplaceAll(image, defaultRegistry, endpoint)
+	replacer := strings.NewReplacer(defaultRegistry, endpoint, packageProdLocation, endpoint, packageDevLocation, endpoint)
+	localImage := replacer.Replace(image)
+	localImage = removeDigestReference(localImage)
```
Comment (g-gaston) [inline pkg/executables/docker.go:89]: I would vote to do this in the registry mover

File: pkg/executables/docker.go [hunk 3]
```diff
@@ -80,7 +84,9 @@ func (d *Docker) CgroupVersion(ctx context.Context) (int, error) {
 }
 
 func (d *Docker) TagImage(ctx context.Context, image string, endpoint string) error {
-	localImage := strings.ReplaceAll(image, defaultRegistry, endpoint)
+	replacer := strings.NewReplacer(defaultRegistry, endpoint, packageProdLocation, endpoint, packageDevLocation, endpoint)
```
Comment (g-gaston) [inline pkg/executables/docker.go:87]: Can we do this in the registry mover as well? That place already has knowledge about the different registries and it's doing some host/path manipulation

File: pkg/executables/docker.go [hunk 4]
```diff
@@ -89,7 +95,9 @@ func (d *Docker) TagImage(ctx context.Context, image string, endpoint string) er
 }
 
 func (d *Docker) PushImage(ctx context.Context, image string, endpoint string) error {
-	localImage := strings.ReplaceAll(image, defaultRegistry, endpoint)
+	replacer := strings.NewReplacer(defaultRegistry, endpoint, packageProdLocation, endpoint, packageDevLocation, endpoint)
+	localImage := replacer.Replace(image)
+	localImage = removeDigestReference(localImage)
```
Comment (g-gaston) [inline pkg/executables/docker.go:100]: Same, move this to the registry mover

File: pkg/executables/docker.go [hunk 5]
```diff
@@ -89,7 +95,9 @@ func (d *Docker) TagImage(ctx context.Context, image string, endpoint string) er
 }
 
 func (d *Docker) PushImage(ctx context.Context, image string, endpoint string) error {
-	localImage := strings.ReplaceAll(image, defaultRegistry, endpoint)
+	replacer := strings.NewReplacer(defaultRegistry, endpoint, packageProdLocation, endpoint, packageDevLocation, endpoint)
```
Comment (g-gaston) [inline pkg/executables/docker.go:98]: same comment about doing this in the registry mover

File: pkg/curatedpackages/reader.go [hunk 6]
```diff
@@ -3,53 +3,79 @@ package curatedpackages
...
 	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
 )
 
+// Temporary: Curated packages dev and prod accounts are currently hard coded
+// This is because there is no mechanism to extract these values as of now
 const (
-	defaultRegistry = "783794618700.dkr.ecr.us-west-2.amazonaws.com"
+	publicProdECR       = "public.ecr.aws/eks-anywhere"
+	packageProdLocation = "783794618700.dkr.ecr.us-west-2.amazonaws.com"
+	packageDevLocation  = "857151390494.dkr.ecr.us-west-2.amazonaws.com"
```
Comment (g-gaston) [inline pkg/curatedpackages/reader.go:20]: for this two, maybe `package[Prod/Dev]Domain`?

File: pkg/curatedpackages/reader.go [hunk 7]
```diff
@@ -61,14 +87,51 @@ func fetchPackages(ctx context.Context, versionsBundle releasev1.VersionsBundle,
...
 	images := make([]releasev1.Image, 0, len(bundle.Spec.Packages))
 	for _, p := range bundle.Spec.Packages {
-		pI := releasev1.Image{
+		pHC := releasev1.Image{
 			Name:        p.Name,
 			Description: p.Name,
 			OS:          ctrl.OS,
 			OSName:      ctrl.OSName,
-			URI:         fmt.Sprintf("%s/%s:%s", defaultRegistry, p.Source.Repository, p.Source.Versions[0].Name),
+			URI:         fmt.Sprintf("%s/%s:%s", GetRegistry(ctrl.URI), p.Source.Repository, p.Source.Versions[0].Name),
```
Comment (g-gaston) [inline pkg/curatedpackages/reader.go:95]: I got lost here. If we are doing some domain manipulation in the image mover, why is this necessary here?

File: pkg/curatedpackages/reader.go [hunk 8]
```diff
@@ -61,14 +87,51 @@ func fetchPackages(ctx context.Context, versionsBundle releasev1.VersionsBundle,
...
+
+	for _, p := range bundle.Spec.Packages {
+		// each package will have at least one version
+		for _, version := range p.Source.Versions[0].Images {
+			image := releasev1.Image{
+				Name:        version.Repository,
+				Description: version.Repository,
+				OS:          ctrl.OS,
+				OSName:      ctrl.OSName,
+				URI:         fmt.Sprintf("%s/%s@%s", getRegistry(ctrl.URI), version.Repository, version.Digest),
```
Comment (g-gaston) [inline pkg/curatedpackages/reader.go:124]: Same question here about why we do this here in addition to what we do in the image mover

File: pkg/manifests/reader.go [hunk 9]
```diff
@@ -80,7 +80,7 @@ func (r *Reader) ReadImages(eksaVersion string) ([]releasev1.Image, error) {
 	return bundles.ReadImages(r, bundle)
 }
 
-func (r *Reader) ReadImagesFromBundles(b *releasev1.Bundles) ([]releasev1.Image, error) {
+func (r *Reader) ReadImagesFromBundles(ctx context.Context, b *releasev1.Bundles) ([]releasev1.Image, error) {
```
Comment (g-gaston) [inline pkg/manifests/reader.go:83]: nit:
```suggestion
func (r *Reader) ReadImagesFromBundles(_ context.Context, b *releasev1.Bundles) ([]releasev1.Image, error) {
```

File: pkg/curatedpackages/reader.go (hunk 7 above)
Comment (a-cool-train) [reply pkg/curatedpackages/reader.go:95]: This is not doing any manipulation. It just extracts the image uri

File: pkg/docker/registry.go (hunk 1 above)
Comment (a-cool-train) [reply pkg/docker/registry.go:18]: This will not be ideal because for dev, the controller will be using l0gr... to fetch images, and if we make this change, it will break the pull of images.

File: pkg/curatedpackages/reader.go (hunk 8 above)
Comment (a-cool-train) [reply pkg/curatedpackages/reader.go:124]: I responded above but this doesn't perform any manipulation, it merely just extracts the images.

File: pkg/executables/docker.go (hunk 3 above)
Comment (a-cool-train) [reply pkg/executables/docker.go:87]: I don't see any host/path manipulation in the registry mover.

File: pkg/executables/docker.go (hunk 5 above)
Comment (a-cool-train) [reply pkg/executables/docker.go]: Added an issue for this: https://github.com/aws/eks-anywhere/issues/3206

File: pkg/executables/docker.go (hunk 3 above)
Comment (a-cool-train) [reply pkg/executables/docker.go]: Added an issue for this: https://github.com/aws/eks-anywhere/issues/3206
